src/main/java/frc/robot/subsystems/Arm.java

```diff
@@ -1,154 +1,307 @@
-    this.leftArmMotor = new CANSparkMax(leftMotorID, MotorType.kBrushless);
-    this.rightArmMotor = new CANSparkMax(rightMotorID, MotorType.kBrushless);
-    leftArmMotor.setIdleMode(IdleMode.kBrake);
-    rightArmMotor.setIdleMode(IdleMode.kBrake);
-    leftArmMotor.setInverted(false);
-    rightArmMotor.setInverted(true);
-    leftArmMotor.setSmartCurrentLimit(ArmConstants.CURRENT_LIMIT);
-    rightArmMotor.setSmartCurrentLimit(ArmConstants.CURRENT_LIMIT);
-    motors = new MotorControllerGroup(leftArmMotor, rightArmMotor);
-
-    this.armEncoder = new DutyCycleEncoder(new DigitalInput(encoderChannel));
-    armEncoder.setDistancePerRotation(ArmConstants.ARM_DEGREES_PER_ROTATION);
-
-    pidController.setTolerance(ArmConstants.ROTATION_PID_TOLERANCE);
+  public Arm(Pivot pivot, Elevator elevator) {
+    this.pivot = pivot;
+    this.elevator = elevator;
   }
 
-  /**
-   * Sets the braking mode to the given {@link IdleMode}.
-   *
-   * @param mode The {@link IdleMode} to set the motors to
-   */
-  public void setIdleMode(IdleMode mode) {
-    leftArmMotor.setIdleMode(mode);
-    rightArmMotor.setIdleMode(mode);
+  public Pivot getPivot() {
+    return this.pivot;
   }
 
-  /**
-   * Gets angle of the arm.
-   * 
-   * @return The rotation of the arm in degrees. 0 = Vertical and pointing down. Positive -> towards front of robot.
-   */
-  public double getAngle() {
-    return ArmConstants.ARM_OFFSET_DEG - armEncoder.getAbsolutePosition() * 360;
+  public Elevator getElevator() {
+    return this.elevator;
   }
 
-  /**
-   * Gets a {@link Rotation2d} representing the current angle of the arm.
-   * 
-   * @return An {@link Rotation2d} of the rotation of the arm. 0 = Vertical and pointing down. Positive -> towards front
-   *         of robot.
-   */
-  public Rotation2d getRotation() {
-    return Rotation2d.fromDegrees(getAngle());
+  public ArmPose getPose() {
+    return ArmPose.fromAngleExtension(this.pivot.getRotation(), this.elevator.getExtension());
   }
 
-  /**
-   * Gets PID setpoint of the arm.
-   * 
-   * @return The setpoint angle of the arm in degrees. 0 = Vertical and pointing down. Positive -> towards front of
-   *         robot.
-   */
-  public double getSetpoint() {
-    return pidController.getSetpoint().position;
+  public ArmPose getSetpointPose() {
+    return ArmPose.fromAngleExtension(Rotation2d.fromDegrees(this.pivot.getSetpoint()),
+        this.elevator.getSetpointExtension());
   }
 
   /**
-   * Sets rotation speed of the arm motors. Includes stop to keep arm from rotating beyond limits.
-   * 
-   * @param speed The desired rotation speed
+   * A sub-subsystem that controls the rotating pivot on the robot.
    */
-  public void setRotationSpeed(double speed) {
-    if (getAngle() < ArmConstants.MIN_ROTATION_DEG && motors.get() < 0
-        || getAngle() > ArmConstants.MAX_ROTATION_DEG && motors.get() > 0) {
-      motors.set(0);
-      return;
+  public static class Pivot {
```

Comment:
Pivot shouldn't need to be static

Reply:
It does, `static` on a class is to allow a nested class to be instantiated independent of any instance of its parent.

Reply:
can we just avoid instantiating separately?